### PR TITLE
HBASE-23549 Document steps to disable MOB for a column family

### PR DESCRIPTION
**Pull request by busbey**
building on #785 (first 5 commits are that PR) to provide a set of steps for disabling the MOB feature safely, for those folks who don't have much data already written into a MOB family and have the HDFS IO to handle the data being in the normal maintenance path.

review here should only be on the doc change "Moving a column family out of MOB". I'll rebase once #785 lands.

### CODE REVIEW

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 53s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  0s |  No case conflicting files found.  |
| +1 :green_heart: |  hbaseanti  |   0m  0s |  Patch does not have any anti-patterns.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
| -0 :warning: |  test4tests  |   0m  0s |  The patch doesn't appear to include any new or modified tests. Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch.  |
||| _ master Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 39s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   6m 53s |  master passed  |
| +1 :green_heart: |  compile  |   3m 52s |  master passed  |
| +1 :green_heart: |  checkstyle  |   3m 16s |  master passed  |
| +0 :ok: |  refguide  |   7m  6s |  branch has no errors when building the reference guide. See footer for rendered docs, which you should manually inspect.  |
| +1 :green_heart: |  shadedjars  |   5m 32s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |   3m 48s |  master passed  |
| +0 :ok: |  spotbugs  |   3m 48s |  Used deprecated FindBugs config; considering switching to SpotBugs.  |
| +1 :green_heart: |  findbugs  |  18m 24s |  master passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 16s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   4m 59s |  the patch passed  |
| +1 :green_heart: |  compile  |   3m  0s |  the patch passed  |
| +1 :green_heart: |  javac  |   3m  0s |  the patch passed  |
| -1 :x: |  checkstyle  |   2m 30s |  root: The patch generated 2 new + 2 unchanged - 0 fixed = 4 total (was 2)  |
| -1 :x: |  whitespace  |   0m  0s |  The patch 2 line(s) with tabs.  |
| +0 :ok: |  refguide  |   5m 27s |  patch has no errors when building the reference guide. See footer for rendered docs, which you should manually inspect.  |
| +1 :green_heart: |  shadedjars  |   5m 22s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  hadoopcheck  |  16m 40s |  Patch does not cause any errors with Hadoop 2.8.5 2.9.2 or 3.1.2.  |
| +1 :green_heart: |  javadoc  |   3m 45s |  the patch passed  |
| +1 :green_heart: |  findbugs  |  18m 49s |  the patch passed  |
||| _ Other Tests _ |
| -1 :x: |  unit  | 176m 41s |  root in the patch failed.  |
| +1 :green_heart: |  asflicense  |   1m 45s |  The patch does not generate ASF License warnings.  |
|  |   | 296m 55s |   |


| Reason | Tests |
|-------:|:------|
| Failed junit tests | hadoop.hbase.io.hfile.TestCacheOnWrite |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=19.03.5 Server=19.03.5 base: https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-928/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/928 |
| Optional Tests | dupname asflicense javac javadoc unit spotbugs findbugs shadedjars hadoopcheck hbaseanti checkstyle compile refguide |
| uname | Linux f8100980e391 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | /home/jenkins/jenkins-slave/workspace/HBase-PreCommit-GitHub-PR_PR-928/out/precommit/personality/provided.sh |
| git revision | master / 72ab1c1193 |
| Default Java | 1.8.0_181 |
| refguide | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-928/1/artifact/out/branch-site/book.html |
| checkstyle | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-928/1/artifact/out/diff-checkstyle-root.txt |
| whitespace | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-928/1/artifact/out/whitespace-tabs.txt |
| refguide | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-928/1/artifact/out/patch-site/book.html |
| unit | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-928/1/artifact/out/patch-unit-root.txt |
|  Test Results | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-928/1/testReport/ |
| Max. process+thread count | 4354 (vs. ulimit of 10000) |
| modules | C: hbase-server hbase-mapreduce . U: . |
| Console output | https://builds.apache.org/job/HBase-PreCommit-GitHub-PR/job/PR-928/1/console |
| versions | git=2.11.0 maven=2018-06-17T18:33:14Z) findbugs=3.1.11 |
| Powered by | Apache Yetus 0.11.1 https://yetus.apache.org |


This message was automatically generated.

**Comment by busbey** [comment]
the checkstyle and whitespace stuff is from #785 and the unit test failures are not related.

File: src/main/asciidoc/_chapters/hbase_mob.adoc [hunk 1]
```diff
@@ -198,3 +198,273 @@ hbase> major_compact 't1', 'c1’, ‘MOB’
...
+values.
+
+The following procedure will safely migrate the underlying data without requiring a cluster outage.
+Clients will see a number of retries when configuration settings are applied and regions are
+reloaded.
+
+.Procedure: Stop MOB maintenance, change MOB threshold, rewrite data via compaction
+. Ensure the MOB compaction chore in the Master is off by setting
+`hbase.mob.file.compaction.chore.period` to `0`. Applying this configuration change will require a
+rolling restat of HBase Masters. That will require at least one fail-over of the active master,
```
[the remark this answers is not in the document]
Comment (busbey) [reply src/main/asciidoc/_chapters/hbase_mob.adoc]: okay to fix on commit if there isn't something else to push a commit for?

[the remark this answers is not in the document]
Comment (petersomogyi) [reply src/main/asciidoc/_chapters/hbase_mob.adoc]: Definitely. Go for it.

**Pull request closed without merging**